kilkari/src/main/java/org/motechproject/nms/kilkari/domain/Subscription.java

```diff
@@ -79,15 +97,74 @@ public void setDeactivationReason(DeactivationReason deactivationReason) {
         this.deactivationReason = deactivationReason;
     }
 
+    public boolean getNeedsWelcomeMessage() {
+        return needsWelcomeMessage;
+    }
+
+    public void setNeedsWelcomeMessage(boolean needsWelcomeMessage) {
+        this.needsWelcomeMessage = needsWelcomeMessage;
+    }
+
     /**
-     * Returns the weekId corresponding to today's date.
+     * Helper method to be called by the OBD process when selecting a message to play for a subscription
+     * @param date The date on which the message will be played
+     * @return SubscriptionPackMessage with the details of the message to play
      */
-    @Ignore
-    int todaysWeekId() {
-        //todo: the work
-        return 0;
+    public SubscriptionPackMessage nextScheduledMessage(DateTime date) { //NO CHECKSTYLE CyclomaticComplexity
```

Comment:
I know that turning off the checkstyle CyclomaticComplexity check is kind of gross... want to get Frank unblocked, can look at simplifying this later if we care.